code/modules/power/lighting.dm

```diff
@@ -559,13 +559,59 @@
 
 /obj/machinery/light/attackby(obj/item/W, mob/user)
 
-	if (issilicon(user) && !isghostdrone(user))
+	if (issilicon(user) && !isghostdrone(user) && !istype(W, /obj/item/borg_lamp_manufacturer))
```

Comment:
Prefer inverting the order of checks rather than doing the `borg_lamp_manufacturer` twice.

Basically:
```
if (istype(W, /obj/item/borg_lamp_manufacturer))
  // stuff
  return
if (issilicon(user) && !isghostdrone(user))
  return
// magtractor stuff
```